nimbus/src/nimbus.idl

```diff
@@ -87,12 +87,31 @@ interface NimbusClient {
     [Throws=Error]
     sequence<EnrollmentChangeEvent> set_global_user_participation(boolean opt_in);
 
-    // Updates the list of experiments from the server. After calling this, the
-    // list of active experiments might change (there might be new experiments,
-    // or old experiments might have expired)
+    // Updates the list of experiments from the server.
+    // This method is deprecated, in favour of calling `fetch_experiments()` and then
+    // `apply_pending_updates()`.
     [Throws=Error]
     sequence<EnrollmentChangeEvent> update_experiments();
 
+    // Fetches the list of experiments from the server. This does not affect the list
+    // of active experiments or experiment enrolment.
+    // Fetched experiments are not applied until `apply_pending_updates()` is called.
+    [Throws=Error]
+    void fetch_experiments();
+
+    // Apply the updated experiments from the last fetch.
```

Comment:
Naming nits, and I apologize for not bringing these up in your doc. I'm also not going to die on this hill :) But:

In `apply_pending_updates()`, the `pending` and `updates` part are somewhat redundant. I wonder if `apply_pending_experiments()` is a better name? ie, you "fetch experiments", then you have "pending experiments" which you can apply.

Reply:
I like your suggestion of removing `updates`, and settled for `experiments` everywhere.